Source/qol/itemlabels.cpp

```diff
@@ -0,0 +1,163 @@
+/**
+* @file itemlabels.cpp
+*
+* Adds item labels QoL feature
+*/
+
+#include <vector>
+#include <string>
+#include <unordered_set>
+
+#include "inv.h"
+#include "gmenu.h"
+#include "cursor.h"
+#include "common.h"
+#include "control.h"
+#include "itemlabels.h"
+#include "utils/language.h"
+#include "engine/render/cel_render.hpp"
+
+namespace devilution {
+
+namespace {
+
+struct itemLabel {
+	int id, width;
+	Point pos;
+	std::string text;
+};
+
+std::vector<itemLabel> labelQueue;
+
+bool altPressed = false;
+bool isLabelHighlighted = false;
+std::array<std::optional<int>, ITEMTYPES> labelCenterOffsets;
+bool invertHighlightToggle = false;
+
+} // namespace
+
+void ToggleItemLabelHighlight()
+{
+	invertHighlightToggle = !invertHighlightToggle;
+}
+
+void AltPressed(bool pressed)
+{
+	altPressed = pressed;
+}
+
+bool IsItemLabelHighlighted()
+{
+	return isLabelHighlighted;
+}
+
+bool IsHighlightingLabelsEnabled()
+{
+	return altPressed != invertHighlightToggle;
+}
+
+void AddItemToLabelQueue(int id, int x, int y)
+{
+	if (!IsHighlightingLabelsEnabled())
+		return;
+	ItemStruct *it = &items[id];
+
+	const char *textOnGround;
+	if (it->_itype == ITYPE_GOLD) {
+		std::sprintf(tempstr, _("%i gold"), it->_ivalue);
+		textOnGround = tempstr;
+	} else {
+		textOnGround = it->_iIdentified ? it->_iIName : it->_iName;
+	}
+
+	int nameWidth = GetLineWidth(textOnGround);
+	BYTE index = ItemCAnimTbl[it->_iCurs];

```

Comment:
Avoid WinAPI types
```suggestion
	uint8_t index = ItemCAnimTbl[it->_iCurs];
```

Reply:
Actually doesn't look like there is a reason to have a specific size here.
```suggestion
	int index = ItemCAnimTbl[it->_iCurs];
```